pkg/serving/config_changes.go

```diff
@@ -331,14 +333,37 @@ func UpdateContainerArg(template *servingv1.RevisionTemplateSpec, arg []string)
 	return nil
 }
 
-// UpdateContainerPort updates container with a give port
-func UpdateContainerPort(template *servingv1.RevisionTemplateSpec, port int32) error {
+// UpdateContainerPort updates container with a given name:port
+func UpdateContainerPort(template *servingv1.RevisionTemplateSpec, port string) error {
 	container, err := ContainerOfRevisionTemplate(template)
 	if err != nil {
 		return err
 	}
+	var containerPort int64
+	var name string
+
+	if strings.Contains(port, ":") {
```

Comment:
To simplify the algorithm, you don't have to check for a colon here, because that already happens when you do the split.

So my suggestions would be (simplified)


```golang
elements := strings.SplitN(port, ":", 2);
if len(elements) == 3 {
   name = elements[0]
   port = parseInt(elements[1])
} else {
  name = ""
  port = parseInt(elements[0])
}
```

and then checking if parseInt doesn't return an error.

Just a cosmetic suggestion which might be easier to follow (and has a smaller [Cyclomatic Complexity](https://en.wikipedia.org/wiki/Cyclomatic_complexity))

Feel free to change it or keep your current implementation.


```

Reply:
@rhuss  this is great. Thanks for these suggestions.

Reply:
if it looks good , i can unhold